### ISSUE TITLE
remote: use local cache as default remote

### ISSUE THREAD

**Issue by dashohoxha** [iterative/dvc#2601]
I will try to explain the idea with a simple example, which can be constructed by this script:
```bash
#!/bin/bash
set -o verbose

rm -rf proj1
mkdir proj1
cd proj1

git init
dvc init -q
git commit -m 'Init DVC'

dvc remote add -d self $(pwd)/.dvc/cache
git add .
git commit -m 'Set default remote'

echo test1 > file
dvc add file
git add .
git commit -m 'test1'
git tag v1

echo test2 > file
dvc commit -f file.dvc
git add .
git commit -m 'test2'
git tag v2

echo test3 > file
dvc commit -f file.dvc
git add .
git commit -m 'test3'
git tag v3

dvc status
git log --oneline
```

Now let's try these:
```bash
cd proj1/
dvc remote list
ls
dvc get . file --rev=v1 -o file.v1
dvc get . file --rev=v2 -o file.v2
ls
cat file
cat file.v1
cat file.v2
```

So, using `dvc get` and its `--rev` option we can get previous versions of a file, and this seems cool to me (at least it is easier than the traditional `git checkout` + `dvc checkout` way).

However `dvc get` requires a **default remote**, otherwise it doesn't work. In the example above I have "tricked" it by using the local cache as a default remote:
```bash
dvc remote add -d self $(pwd)/.dvc/cache
```
Without this trick it would have failed.

So, I was thinking, what if DVC uses **by default** the local cache as a _default remote_?
This could also help in other cases when a default remote is required and the command is failing because of this (one of these cases might be this one: https://github.com/iterative/dvc/issues/2599).

**Comment by shcheklein** [comment]
I think it should not be failing if remote is not specified _and_ file is in the cache already. Not sure if we need to treat local cache as a default remote though. What would be semantics of `dvc push`? I think it can be confusing for people if it will be working even when remote is not defined.

**Comment by jorgeorpinel** [comment]
Was this closed by #2915 perhaps?

**Comment by efiop** [comment]
@jorgeorpinel Yes, thank you!
